.github/workflows/android.yml

```diff
@@ -38,3 +38,35 @@ jobs:
             plugins/gradle/build/reports/tests/test
             if-no-files-found: ignore
             retention-days: 15
+
+  validateAGP7:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        agp: ['7.1.0','7.1.1','7.1.2','7.1.3','7.2.0','7.2.1','7.2.2','7.3.0','7.3.1','7.4.0','7.4.1','7.4.2']
+        
+    name: AGP Matrix Test - AGP ${{ matrix.agp }}
+    env:
+      VERSION_AGP: ${{ matrix.agp }}
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Validate with JDK 11
+      uses: actions/setup-java@v3
+      with:
+        java-version: '11'
+        distribution: 'temurin'
+        cache: gradle
+    - name: Set up Gradle
+      uses: gradle/gradle-build-action@v2.4.2
+      with:
+        gradle-version: 7.2
+    - name: Grant execute permission for gradlew
+      run: chmod +x gradlew
+    - name: Stage with Gradle
+      run: ./gradlew install publish
+    - name: Grant execute permission for gradlew
+      run: chmod +x gradlew
+    - name: Build with Gradle
+      run: ./gradlew clean build
```

Comment:
Your matriv values should be passed to the build as properties (-P). SO for example,
```./gradlew -Pnewrelic.agent.version=7.+ -Pnewrelic.agp.version=${{ matrix.agp }} clean build 
```